build/scripts-3.11/import.py

```diff
@@ -0,0 +1,191 @@
+#!python
+
+import os
+import sys
+import ssl
+import glob
+import yaml
+import logging
+import argparse as ap
+import collections as col
+import dpimport.importer as importer
+from dpimport.database import Database
+from pymongo import DeleteMany, UpdateMany
+from pymongo.errors import BulkWriteError
+
+logger = logging.getLogger(__name__)
+
+def main():
+    parser = ap.ArgumentParser()
+    parser.add_argument('-c', '--config')
+    parser.add_argument('-d', '--dbname', default='dpdata')
+    parser.add_argument('-v', '--verbose', action='store_true')
+    parser.add_argument('expr')
+    args = parser.parse_args()
+
+    level = logging.INFO
+    if args.verbose:
+        level = logging.DEBUG
+    logging.basicConfig(level=level)
+
+    with open(os.path.expanduser(args.config), 'r') as fo:
+        config = yaml.load(fo, Loader=yaml.SafeLoader)
+
+    # db = Database(config, args.dbname).connect()
+
+    # iterate over matching files on the filesystem
+    for f in glob.iglob(args.expr, recursive=True):
+        print(f)
+    #     dirname = os.path.dirname(f)
+    #     basename = os.path.basename(f)
+    #     # probe for dpdash-compatibility and gather information
+    #     probe = dpimport.probe(f)
+    #     if not probe:
+    #         logger.debug('document is unknown %s', basename)
+    #         continue
+    #     # nothing to be done
+    #     if db.exists(probe):
+    #         logger.info('document exists and is up to date %s', probe['path'])
+    #         continue
+    #     logger.info('document does not exist or is out of date %s', probe['path'])
+    #     # import the file
+    #     logger.info('importing file %s', f)
+    #     dppylib.import_file(db.db, probe)
+
+    # logger.info('cleaning metadata')
+    # lastday = get_lastday(db.db)
+    # if lastday:
+    #     clean_metadata(db.db, lastday)
+
+def clean_metadata(db, max_days):
+    studies = col.defaultdict()
+    subjects = list()
+
+    for subject in max_days:
+        if subject['_id']['study'] not in studies:
+            studies[subject['_id']['study']] = {}
+            studies[subject['_id']['study']]['subject'] = []
+            studies[subject['_id']['study']]['max_day'] = 0
```

Comment:
`subject['_id']['study']` is referenced a lot in this loop. Feels like it would be worth it to both give it a clear name/reason and also to reduce noise on any given line.

Reply:
[655c34d](https://github.com/AMP-SCZ/dpimport/pull/17/commits/655c34d7cc922806cb0928f4d5234932a18bcfaf)